ecommerce/extensions/iap/api/v1/views.py

```diff
@@ -0,0 +1,95 @@
+from edx_rest_framework_extensions.permissions import LoginRedirectIfUnauthenticated
+from oscar.apps.basket.views import *  # pylint: disable=wildcard-import, unused-wildcard-import
+from rest_framework.views import APIView
+from ecommerce.extensions.analytics.utils import track_segment_event
+from ecommerce.extensions.basket.constants import EMAIL_OPT_IN_ATTRIBUTE
+from ecommerce.extensions.basket.exceptions import BadRequestException, RedirectException
+from ecommerce.extensions.basket.utils import prepare_basket
+from ecommerce.extensions.basket.views import BasketLogicMixin
+from ecommerce.extensions.order.exceptions import AlreadyPlacedOrderException
+from ecommerce.extensions.partner.shortcuts import get_partner_for_site
+
+BasketAttribute = get_model('basket', 'BasketAttribute')
+BasketAttributeType = get_model('basket', 'BasketAttributeType')
+logger = logging.getLogger(__name__)
+Product = get_model('catalogue', 'Product')
+Voucher = get_model('voucher', 'Voucher')
+
+
+class MobileBasketAddItemsView(BasketLogicMixin, APIView):
+    """
+    View that adds multiple products to a mobile user's basket.
+    """
+    permission_classes = (LoginRedirectIfUnauthenticated,)
+
+    def get(self, request):
+        # Send time when this view is called - https://openedx.atlassian.net/browse/REV-984
+        properties = {'emitted_at': time.time()}
+        track_segment_event(request.site, request.user, 'Basket Add Items View Called', properties)
+
+        try:
+            skus = self._get_skus(request)
+            products = self._get_products(request, skus)
+
+            logger.info('Starting payment flow for user [%s] for products [%s].', request.user.username, skus)
+
+            available_products = self._get_available_products(request, products)
+
+            try:
+                basket = prepare_basket(request, available_products)
+            except AlreadyPlacedOrderException:
+                return JsonResponse({'error': _('You have already purchased these products')}, status=406)
+
+            self._set_email_preference_on_basket(request, basket)
+
+            return JsonResponse({'success': _('Course added to the basket successfully'), 'basket_id': basket.id}, status=200)
+
+        except BadRequestException as e:
+            return JsonResponse({'error': str(e)}, status=400)
+        except RedirectException as e:
+            return e.response
+
+    def _get_skus(self, request):
+        skus = [escape(sku) for sku in request.GET.getlist('sku')]
+        if not skus:
+            raise BadRequestException(_('No SKUs provided.'))
+        return skus
+
+    def _get_products(self, request, skus):
+        partner = get_partner_for_site(request)
+        products = Product.objects.filter(stockrecords__partner=partner, stockrecords__partner_sku__in=skus)
+        if not products:
+            raise BadRequestException(_('Products with SKU(s) [{skus}] do not exist.').format(skus=', '.join(skus)))
```

Comment:
Should this be a 404 instead?

Also, it's not clear from the PR description testing instructions that multiple SKUs can be added in a single call.  Can you make sure this is included in a spec, and the testing instructions updated?

Reply:
Sure I will add more details in the description. They are infact possible even in the original code/flow